### PR TITLE
PDP - Single and Bundle PDP SMK-31, SMK-33

### PR DESCRIPTION
**Pull request by emad-tabib**

### CODE REVIEW

File: src/com/generic/page/HomePage.java [hunk 1]
```diff
@@ -39,6 +39,12 @@ public static void NavigateAwayFromHomePage() throws Exception {
 		getCurrentFunctionName(false);
 	}
 
+	public static void disableMonetate() throws Exception {
+		getCurrentFunctionName(true);
+		getDriver().get(getURL() + "/?monetate=off");
+		getCurrentFunctionName(false);
+	}
+
```
Comment (ibatta) [inline src/com/generic/page/HomePage.java]: Do we really needto do this  ?

File: src/com/generic/page/PDP.java [hunk 2]
```diff
@@ -47,19 +52,29 @@
...
+//	}
 
 	// done - SMK
-	// to be replaced with Search
-	public static void NavigateToGRPDP() throws Exception {
+	public static void NavigateToPDP(String SearchTerm) throws Exception {
 		getCurrentFunctionName(true);
-		getDriver().get(getURL() + keys.GR_PDP);
+		if (SelTestCase.getBrowserName().contains(GlobalVariables.browsers.iPad))
+			HomePage.disableMonetate();
```
Comment (ibatta) [inline src/com/generic/page/PDP.java]: we need to make this tempo. operation  
Please add this to configurations file

File: FG/com/generic/tests/FG/PDP/PDPValidation.java [hunk 3]
```diff
@@ -1,22 +1,37 @@
...
 
 public class PDPValidation extends SelTestCase {
 
-	public static void validate() throws Exception {
+	public static void validate(String search) throws Exception {
 		getCurrentFunctionName(true);
+		PDP.NavigateToPDP(search);
+		int numberOfItems = PDP.getNumberOfItems();
+		String priceErrorMessage;
+		if (PDP.getNumberOfItems() == 1) {
```
Comment (ibatta) [inline FG/com/generic/tests/FG/PDP/PDPValidation.java]: replace it with  numberOfItems

File: FG/com/generic/tests/FG/PDP/PDPValidation.java [hunk 4]
```diff
@@ -1,22 +1,37 @@
...
+			priceErrorMessage = "Top price is not dispayed";
```
Comment (ibatta) [inline FG/com/generic/tests/FG/PDP/PDPValidation.java]: add comment here to explain Please explain why?

File: src/com/generic/page/PDP.java [hunk 5]
```diff
@@ -30,7 +30,12 @@
 	public static class keys {
 		//to be replaced with Search
 		public static String FG_PDP = "/cheyne-high-low-area-rug/915951";
+		public static String FG_PDP2 = "/barrow-chesterfield-sofa/691936";
+		public static String FG_BundlePDP = "/resort-cotton-bath-towels/155771";
+		public static String FG_BundlePDP2 = "/premium-monogrammed-disposable-napkins/605617?brReference=BRWidget";
```
Comment (ibatta) [inline src/com/generic/page/PDP.java]: Let us find a better way to do this, may be we need to pull them all from the data sheet

File: GR/com/generic/tests/GR/PDP/PDPValidation.java [hunk 6]
```diff
@@ -1,22 +1,40 @@
 package com.generic.tests.GR.PDP;
 
 import com.generic.page.PDP;
+import com.generic.setup.GlobalVariables;
 import com.generic.setup.SelTestCase;
 
 public class PDPValidation extends SelTestCase {
 
-	public static void validate() throws Exception {
+	public static void validate(String search) throws Exception {
```
Comment (ibatta) [inline GR/com/generic/tests/GR/PDP/PDPValidation.java]: call it searchTerm

File: GR/com/generic/tests/GR/PDP/PDPValidation.java [hunk 7]
```diff
@@ -1,22 +1,40 @@
...
 public class PDPValidation extends SelTestCase {
 
-	public static void validate() throws Exception {
+	public static void validate(String search) throws Exception {
 		getCurrentFunctionName(true);
 
-		PDP.NavigateToGRPDP();
-		sassert().assertTrue(PDP.validatePriceIsDisplayed(), "Top price is not dispayed");
+
+		PDP.NavigateToPDP(search);
```
Comment (ibatta) [inline GR/com/generic/tests/GR/PDP/PDPValidation.java]: same previous comment

File: src/com/generic/page/PDP.java [hunk 8]
```diff
@@ -260,58 +303,104 @@ public static void addProductsToCart() throws Exception {
...
 		// because there is no attribute to verify if it is enabled.
-		SelectorUtil.isDisplayed(PDPSelectors.addToWLGRBtnEnabled.get());
+		String selectorEnabled = PDPSelectors.addToWLGRBtnEnabledSingle.get();
+		String selectorDisabled = PDPSelectors.addToCartBtnDisabledSingle.get();
+		if (getNumberOfItems() > 1 && !SelTestCase.getBrowserName().contains(GlobalVariables.browsers.iPhone)) {
+			String ProductID = getProductID(0);
+			selectorEnabled = "css,#" + ProductID + ">"
+					+ PDPSelectors.addToWLGRBtnEnabledSingle.get().replace("css,", "");
+			selectorDisabled = "css,#" + ProductID + ">"
+					+ PDPSelectors.addToCartBtnDisabledSingle.get().replace("css,", "");
```
Comment (ibatta) [inline src/com/generic/page/PDP.java]: Please create new selectors on PDP selecrtors using same method you did here

File: src/com/generic/page/PDP.java [hunk 9]
```diff
@@ -260,58 +303,104 @@ public static void addProductsToCart() throws Exception {
...
-		SelectorUtil.isDisplayed(PDPSelectors.addToCartBtnEnabled.get());
+		String selectorEnabled = PDPSelectors.addToCartBtnEnabledSingle.get();
+		String selectorDisabled = PDPSelectors.addToCartBtnDisabledSingle.get();
+		if (getNumberOfItems() > 1 && !SelTestCase.getBrowserName().contains(GlobalVariables.browsers.iPhone)) {
+			String ProductID = getProductID(0);
+			selectorEnabled = "css,#" + ProductID + ">"
+					+ PDPSelectors.addToCartBtnEnabledSingle.get().replace("css,", "");
+			selectorDisabled = "css,#" + ProductID + ">"
+					+ PDPSelectors.addToCartBtnDisabledSingle.get().replace("css,", "");
+		}
```
Comment (ibatta) [inline src/com/generic/page/PDP.java]: same comment

File: src/com/generic/page/PDP.java [hunk 10]
```diff
@@ -333,6 +422,234 @@ public static void clickAddToWLGR() throws Exception {
...
+	public static void selectSwatches() throws Exception {
+		try {
+			getCurrentFunctionName(true);
+			if (getNumberOfItems() > 1 && !SelTestCase.getBrowserName().contains(GlobalVariables.browsers.iPhone)) {
+				String ProductID = getProductID(0);
+				String ListSelector = "css,#" + ProductID + ">" + PDPSelectors.ListBox.get().replace("css,", "");
+				String activeLists = "css,#" + ProductID + ">" + PDPSelectors.activeListBox.get().replace("css,", "");
+				String swatchContainerSelector = "css,#" + ProductID + ">" + PDPSelectors.swatchContainer.get().replace("css,", "");
+				String imageOptionSelector = "css,#" + ProductID + ">" + PDPSelectors.imageOption.get().replace("css,", "");
+				int numberOfActiveListBoxes = 0;
```
Comment (ibatta) [inline src/com/generic/page/PDP.java]: same

File: src/com/generic/page/PDP.java [hunk 11]
```diff
@@ -333,6 +422,234 @@ public static void clickAddToWLGR() throws Exception {
...
+						} else {
+							break;
+						}
+					}
+				}
+				int numberOfimageOptions = getNumberOfimageOptionsInProduct(swatchContainerSelector);
+				if (numberOfimageOptions != 0) {
+					for (index ++ ; index <= numberOfimageOptions + listIndex; index++) {
+						selectNthOptionFirstSwatchV2("css,#" + ProductID + ">"
+								+ MessageFormat.format(PDPSelectors.imageOption.get(), index).replace("css,", ""));
```
Comment (ibatta) [inline src/com/generic/page/PDP.java]: same

File: FG/com/generic/tests/FG/PDP/PDPValidation.java (hunk 4 above)
Comment (emad-tabib) [reply FG/com/generic/tests/FG/PDP/PDPValidation.java]: Done, thanks

File: src/com/generic/page/HomePage.java (hunk 1 above)
Comment (emad-tabib) [reply src/com/generic/page/HomePage.java]: yes, to disable the Monetate on search field.

File: FG/com/generic/tests/FG/PDP/PDPValidation.java (hunk 3 above)
Comment (emad-tabib) [reply FG/com/generic/tests/FG/PDP/PDPValidation.java]: Done

File: src/com/generic/page/PDP.java (hunk 5 above)
Comment (emad-tabib) [reply src/com/generic/page/PDP.java]: Removed, no longer needed

**Pull request merged**